chaco/colormapped_selection_overlay.py

```diff
@@ -102,24 +103,43 @@ def overlay(self, component, gc, view_bounds=None, mode="normal"):
 
     def _component_changed(self, old, new):
         if old:
-            old.on_trait_change(self.datasource_change_handler, "color_data", remove=True)
+            old.observe(self.datasource_change_handler, "color_data", remove=True)
         if new:
-            new.on_trait_change(self.datasource_change_handler, "color_data")
+            new.observe(self.datasource_change_handler, "color_data")
             self._old_alpha = new.fill_alpha
             self._old_outline_color = new.outline_color
             self._old_line_width = new.line_width
-            self.datasource_change_handler(new, "color_data", None, new.color_data)
+            
+            self.datasource_change_handler(
+                TraitChangeEvent(
+                    object=new,
+                    name="color_data",
+                    old=None,
+                    new=new.color_data
+                )
+            )
```

Comment:
This is certainly a pattern we would like to avoid.  For now I just went with the most straightforward transition possible to preserve behavior, but at some point we should rethink this